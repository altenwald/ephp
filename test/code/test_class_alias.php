<?php

class foo { }

class_alias('foo', 'bar');

$a = new foo;
$b = new bar;

var_dump([
	// objects are the same
	"a == b" => $a == $b,
	"a === b" => $a === $b,
	"a instance of b" => $a instanceof $b,

	// classes are the same
	"a instance of foo" => $a instanceof foo,
	"a instance of bar" => $a instanceof bar,
	"b instance of foo" => $b instanceof foo,
	"b instance of bar" => $b instanceof bar,
]);

