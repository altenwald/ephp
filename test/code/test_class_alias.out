array(7) {
  ["a == b"]=>
  bool(true)
  ["a === b"]=>
  bool(false)
  ["a instance of b"]=>
  bool(true)
  ["a instance of foo"]=>
  bool(true)
  ["a instance of bar"]=>
  bool(true)
  ["b instance of foo"]=>
  bool(true)
  ["b instance of bar"]=>
  bool(true)
}
